docs/source/developer/distributions.rst

```diff
@@ -0,0 +1,232 @@
+
+Continuous submodule
+~~~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.continuous` module contains the classes used for continuous distributions.
+
+DateTime submodule
+~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.datetime` module contains the classes used for ``DateTime`` distributions.
+
+Discrete submodule
+~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.discrete` module contains the classes used to for discrete distributions.
+
+Faker submodule
+~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.faker` module contains the classes used for distributions that are based on the `Faker <https://faker.readthedocs.io/en/master/>`_ package.
+
+NA submodule
+~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.na` module contains the :class:`metasyn.distribution.NADistribution` class, a distribution which generates *only* NA values.
+
+Regex submodule
+~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.regex` module contains the classes for distributions that are based on regular expressions. It implements the open source `regexmodel <https://github.com/sodascience/regexmodel>`_ package. 
+
+There is also a legacy module :mod:`~metasyn.distribution.legacy.regex` that contains the old implementation of the regex distribution. 
+
+
+Creating a new distribution
+---------------------------
+
+New distributions can be created by either directly inheriting from a base class such as the :class:`~metasyn.distribution.BaseDistribution` (or a specialized base class such as :class:`~metasyn.distribution.ScipyDistribution`)
+
+For example, let's say we want to implement a new series of similar distributions, we can create a base class inheriting from :class:`~metasyn.distribution.BaseDistribution`:
+
+.. code-block:: python
+
+    from metasyn.distribution.base import BaseDistribution, metadist
+
+
+    class BaseNewDistribution(BaseDistribution):
+        """Base class for new distribution.
+        This base class makes it easy to implement variations of this distribution for different variable types.
+        """
+
+        # implementation details here
+
+Then we can create distributions for different variable types by inheriting from this base class, and using the ``metadist`` decorator to set the proper attributes. 
+
+.. admonition:: Important
+
+    In cases where there are multiple variations of a distribution for different data types, such as in the example given in this section, or the ``core.uniform`` distributions in ``metasyn``. 
+    
+    In this case, the metasyn convention is to put the data type **after** the distribution name, e.g. ``core.uniform_discrete``, ``core.uniform_datetime``, ``core.uniform_date``, etc.
+
+We can create a new distribution for continuous, discrete and string variables as follows (note that the implementation here is a basic implementation serving as an example):
+
+.. code-block:: python
```

Comment:
I think it might be nicer to have a single example, with all 5 mandatory methods "implemented". Also, we should explain whether you should use "core.*" or "extra.*", "user.*", etc.